src/lightning_lite/plugins/collectives/collective.py

```diff
@@ -0,0 +1,137 @@
+from abc import ABC, abstractmethod
+from typing import Any, List, Optional
+
+import torch
+from typing_extensions import Self
+
+from lightning_lite.utilities.types import CollectibleGroup
+
+
+class Collective(ABC):
+    def __init__(self, instantiate_group: bool = False, **kwargs: Any) -> None:
+        self._kwargs = kwargs
+        self._group: Optional[CollectibleGroup] = None
+        if instantiate_group:
+            self.create_group()
+
+    @property
+    @abstractmethod
+    def rank(self) -> int:
+        ...
+
+    @property
+    @abstractmethod
+    def world_size(self) -> int:
```

Comment:
```suggestion
    def size(self) -> int:
```
or
```suggestion
    def group_size(self) -> int:
```
to be compliant with the semantics of torch groups? Thoughts?

Reply:
torch uses `get_world_size`